panther_lights/src/driver_node.cpp

```diff
@@ -53,67 +52,114 @@ DriverNode::DriverNode(const std::string & node_name, const rclcpp::NodeOptions
-  chanel_2_ts_ = this->get_clock()->now();
-
-  chanel_1_.SetGlobalBrightness(global_brightness);
-  chanel_2_.SetGlobalBrightness(global_brightness);
-
   chanel_1_sub_ = std::make_shared<image_transport::Subscriber>(
     it->subscribe("lights/driver/channel_1_frame", 5, [&](const ImageMsg::ConstSharedPtr & msg) {
       FrameCB(msg, chanel_1_, chanel_1_ts_, "channel_1");
       chanel_1_ts_ = msg->header.stamp;
     }));
 
+  chanel_2_ts_ = this->get_clock()->now();
   chanel_2_sub_ = std::make_shared<image_transport::Subscriber>(
     it->subscribe("lights/driver/channel_2_frame", 5, [&](const ImageMsg::ConstSharedPtr & msg) {
       FrameCB(msg, chanel_2_, chanel_2_ts_, "channel_2");
       chanel_2_ts_ = msg->header.stamp;
     }));
 
-  set_brightness_server_ = this->create_service<SetLEDBrightnessSrv>(
-    "lights/driver/set/brightness", std::bind(&DriverNode::SetBrightnessCB, this, _1, _2));
-
-  RCLCPP_INFO(this->get_logger(), "Initialized successfully.");
+  RCLCPP_INFO(this->get_logger(), "Initialized ImageTransport successfully.");
 }
 
 void DriverNode::OnShutdown()
 {
-  // Clear LEDs
+  ClearLEDs();
+
+  if (led_control_granted_) {
+    ToggleLEDControl(false);
+  }
+}
+
+void DriverNode::ClearLEDs()
+{
   chanel_1_.SetPanel(std::vector<std::uint8_t>(num_led_ * 4, 0));
   chanel_2_.SetPanel(std::vector<std::uint8_t>(num_led_ * 4, 0));
+}
 
-  // Give back control over LEDs
-  if (panels_initialised_) {
-    SetPowerPin(false);
+void DriverNode::ToggleLEDControl(const bool enable)
+{
+  auto request = std::make_shared<SetBoolSrv::Request>();
+  request->data = enable;
+
+  if (!enable_led_control_client_->wait_for_service(std::chrono::seconds(3))) {
+    throw std::runtime_error(
+      "Timeout occurred while waiting for service '" +
+      std::string(enable_led_control_client_->get_service_name()) + "'!");
   }
 
-  gpio_driver_.reset();
+  enable_led_control_client_->async_send_request(
+    request, std::bind(&DriverNode::ToggleLEDControlCB, this, std::placeholders::_1));
+
+  RCLCPP_DEBUG(
+    this->get_logger(), "Sent request toggling LED control to '%s'.", enable ? "true" : "false");
+}
+
+void DriverNode::ToggleLEDControlCB(rclcpp::Client<SetBoolSrv>::SharedFutureWithRequest future)
+{
+  RCLCPP_DEBUG(this->get_logger(), "Received response after toggling LED control.");
+
+  auto result = future.get();
+
+  auto request = result.first;
+  auto response = result.second;
+
+  if (request->data == true && response->success) {
+    led_control_granted_ = true;
+    ClearLEDs();
+    RCLCPP_INFO_THROTTLE(this->get_logger(), *this->get_clock(), 5000, "LED control granted.");
+  } else if (request->data == false && response->success) {
+    led_control_granted_ = false;
+    RCLCPP_INFO_THROTTLE(this->get_logger(), *this->get_clock(), 5000, "LED control revoked.");
+  } else {
+    RCLCPP_ERROR_THROTTLE(
+      this->get_logger(), *this->get_clock(), 5000, "Failed to toggle LED control.");
+  }
```

Comment:
```suggestion
  if (!response->success){
    RCLCPP_ERROR_THROTTLE(
      this->get_logger(), *this->get_clock(), 5000, "Failed to toggle LED control.");
    return;
  }

  if (request->data == true) {
    led_control_granted_ = true;
    ClearLEDs();
    RCLCPP_INFO_THROTTLE(this->get_logger(), *this->get_clock(), 5000, "LED control granted.");
  } else {
    led_control_granted_ = false;
    RCLCPP_INFO_THROTTLE(this->get_logger(), *this->get_clock(), 5000, "LED control revoked.");
  }
```
Personally would do it this way